dev-qt/qt-creator/qt-creator-8.0.0.ebuild

```diff
@@ -0,0 +1,311 @@
+		use l10n_${lang} && languages+=( "${lang/-/_}" )
+	done
+	sed -i -e "s|^set(languages.*|set(languages ${languages[*]})|" \
+		share/qtcreator/translations/CMakeLists.txt || die
+
+	# remove bundled yaml-cpp
+	rm -r src/libs/3rdparty/yaml-cpp || die
+
+	# remove bundled qbs
+	rm -r src/shared/qbs || die
+
+	# conditionally remove bundled qlitehtm
+	if ! use webengine; then
+		# unbundled gumbo doesn't use cmake
+		local gumbo_dep='pkg_check_modules(gumbo REQUIRED IMPORTED_TARGET gumbo)\n'
+		sed -e '/^\s*gumbo/s|gumbo|PkgConfig::gumbo|' \
+			-e "/^find_package(litehtml/s|^|${gumbo_dep}|" \
+			-i src/libs/qlitehtml/src/CMakeLists.txt
+	fi
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_PLUGIN_DESIGNER=$(usex designer)
+		-DBUILD_PLUGIN_IMAGEVIEWER=$(usex imageviewer)
+		-DBUILD_PLUGIN_SERIALTERMINAL=$(usex serialterminal)
+		-DWITH_DEBUG_CMAKE=$(usex debug)
+		-DWITH_TESTS=$(usex test)
+
+		# Docs
+		-DWITH_DOCS=$(usex doc)
+		-DBUILD_DEVELOPER_DOCS=$(usex doc)
+		-DWITH_ONLINE_DOCS=NO # Install failure.  Disable for now
+
+		# Clang stuff
+		-DClang_DIR="${CLANG_PREFIX}/$(get_libdir)/cmake/clang"
+		-DLLVM_DIR="${CLANG_PREFIX}/$(get_libdir)/cmake/llvm"
+		-DCLANGTOOLING_LINK_CLANG_DYLIB=YES
+		-DBUILD_PLUGIN_CLANGCODEMODEL=$(usex clang)
+		-DBUILD_PLUGIN_CLANGFORMAT=$(usex clang)
+		-DBUILD_PLUGIN_CLANGTOOLS=$(usex clang)
+
+		# Help viewer backend stuff
+		-DBUILD_HELPVIEWERBACKEND_QTWEBENGINE=$(usex webengine)
+		-DHELPVIEWER_DEFAULT_BACKEND=$(usex webengine qtwebengine litehtml)
+		-DBUILD_LIBRARY_QLITEHTML=$(usex webengine NO YES)
+
+		# QML stuff
+		-DBUILD_PLUGIN_QMLDESIGNER=$(usex qml)
+		-DBUILD_PLUGIN_QMLJSEDITOR=$(usex qml)
+		-DBUILD_PLUGIN_QMLPREVIEW=$(usex qml)
+		-DBUILD_PLUGIN_QMLPROFILER=$(usex qml)
+		-DBUILD_PLUGIN_QMLPROJECTMANAGER=$(usex qml)
+		-DBUILD_PLUGIN_STUDIOWELCOME=$(usex qml)
+
+		# QT6 can only use bundled version
+		-DBUILD_LIBRARY_KSYNTAXHIGHLIGHTING="$(usex qt6)"
+
+		# An entire mode devoted to a giant "Hello World!" button that does nothing.
+		# TODO: Maybe add to an "examples" USE flag
+		-DBUILD_PLUGIN_HELLOWORLD=NO
+
+		# Not usable in linux environment
+		-DBUILD_PLUGIN_IOS=NO
+
+		# Use portage to update
+		-DBUILD_PLUGIN_UPDATEINFO=NO
+
+		# Don't spam "created by a different GCC executable [-Winvalid-pch]"
+		-DBUILD_WITH_PCH=NO
+
+		# Don't use SANITIZE_FLAGS to pass extra CXXFLAGS
+		-DWITH_SANITIZE=NO
+	)
+
+	use webengine || mycmakeargs+=(
+		-DEXTERNAL_GUMBO=YES
+		-DEXTERNAL_XXD=NO
+		-DLITEHTML_UTF8=YES
+	)
+
+	cmake_src_configure
+}
+
+src_test() {
+	virtx cmake_src_test
+}
+
+src_install() {
+	cmake_src_install
+
+	if use doc; then
+		cmake_src_install doc/{qch,html}_docs
+		docinto  html
+		dodoc -r "${BUILD_DIR}"/doc/html/.
+		insinto /usr/share/qt5-doc # looks here, regardless of qt slot
+		doins "${BUILD_DIR}"/share/doc/qtcreator/*.qch
+	fi
+
+	# Useful scripts that can be used in projects to generate tasklist files
+	insinto /usr/share/qtcreator/scripts
+	doins scripts/*tasks*
```

Comment:
disagree with installing these, I don't see them in the official packages or anywhere else.